### ISSUE TITLE
Add setting for rgba → hex conversion?

### ISSUE THREAD

**Issue by myobie** [disrupted/dprint-plugin-css#3]
The default conversion of `rgba(...)` to a long hex is not ideal for my project. Is this easy to make configurable? 

_(I looked at possibly trying to add the setting on my own, but it's not possible for me right now.)_

**Comment by disrupted** [comment]
This is something I would like myself. Unfortunately it is not that trivial, because it's not supported out of the box by [parcel_css](https://docs.rs/parcel_css/1.0.0-alpha.25/parcel_css/index.html) afaict.

For this to become possible, this plugin first needs to implement custom AST → IR generation, where we can then define our own rules.
At the moment, all I am doing is essentially `StyleSheet::parse` and then `stylesheet.to_css`  [[format_text.rs](https://github.com/disrupted/dprint-plugin-css/blob/main/src/format_text.rs)], so parcel_css takes care of all the formatting decisions.
What we need to do instead, is to parse the [`CSSRule`s](https://docs.rs/parcel_css/1.0.0-alpha.25/parcel_css/rules/struct.CssRuleList.html) inside [`StyleSheet`](https://docs.rs/parcel_css/1.0.0-alpha.25/parcel_css/stylesheet/struct.StyleSheet.html) and make our own formatting decisions (when to insert a line break etc).  

I am hopefully gonna have the time to turn this into a fully featured formatting plugin for dprint at some point, but probably not anytime soon.

If you know a little Rust and are willing to contribute, feel free to to look into it :)

**Comment by disrupted** [comment]
hey @myobie,
I've completely rebuilt this formatter and defined custom rules for the AST → IR transformation. Been kinda chugging along on it for a couple of months whenever I had time and now it's finished.
Please update the plugin to the latest version and give it a try :)

**Comment by myobie** [comment]
@disrupted 🙇  wow that's a big change. I ran this through my codebase and the original big problems I saw are gone:

1. `:global()` just didn't work – it would remove the parens – is fixed
2. rgb/a colors were auto-converted to hex (this issue) – is fixed
3. It would add weird spaces inside complex transform function calls) – is fixed

However, it seems to have removed all my hex color values from my custom properties. An example diff is:

```diff
-  --color: #fff;
-  --fill: #fff;
+  --color: #;
+  --fill: #;
```

It did this to 3 or 6 character hex values. I wrote a failing test for it and push it here: https://github.com/disrupted/dprint-plugin-css/pull/7
